src/ethereum/networkAdapters/EvmScanAdapter.ts

```diff
@@ -367,11 +368,18 @@ export class EvmScanAdapter extends NetworkAdapter<EvmScanAdapterConfig> {
 
     let l1RollupFee = '0'
     if (isSpend && this.ethEngine.networkInfo.optimismRollup === true) {
-      const response = await this.ethEngine.ethNetwork.multicastRpc(
+      const rpcResponse = await this.ethEngine.ethNetwork.multicastRpc(
         'eth_getTransactionReceipt',
         [txid]
       )
-      const json = asGetTransactionReceipt(response.result.result)
+      let json = asMaybe(asGetTransactionReceipt)(rpcResponse.result.result)
+      if (json == null) {
+        const path = `?module=proxy&action=eth_getTransactionReceipt&txhash=${txid}`
+        const evmScanResponse = await this.serialServers(
+          async server => await this.fetchGetEtherscan(server, path)
+        )
+        json = asGetTransactionReceipt(evmScanResponse.result)
+      }
```

Comment:
Why not exclusively use this method to get the receipt since this is a EvmScanAdapter. I suppose it's nice to have a fallback, but perhaps the fallback should be `multicastRpc` instead of the evmscan connection. By removing `multicastRpc` from EvmScanAdapter, we've decoupled the evmscan adapter from the rpc adapter entirely. Is there any reason why we should keep multicastRpc around other then some redundancy?

Reply:
Besides redundancy the other consideration is potentially cost since rpcs are generally cheaper to use but the potential benefit/detriment would vary chain by chain. This method isn't public on the adaptor so I don't think it matters what node list is actually used under the hood.